feature/signupflow/onboarding/src/main/java/com/epmedu/animeal/signup/onboarding/presentation/viewmodel/OnboardingState.kt

```diff
@@ -0,0 +1,7 @@
+package com.epmedu.animeal.signup.onboarding.presentation.viewmodel
+
+import com.epmedu.animeal.signup.onboarding.domain.FacebookAuthorization
+
+data class OnboardingState(
+    val facebookAuthorization: FacebookAuthorization? = null
```

Comment:
I would rather convert `AuthenticationType` to sealed interface, move `isPhoneNumberVerified` field to `Facebook` data class inside the interface and use it in `OnboardingState`. 
This way you can remove redundant `FacebookAuthorization` data class and put all navigation logic into `onState` function (depending on `AuthenticationType` and `isPhoneNumberVerified`). 
Also don't forget to rename `FacebookSignInFinished` event to just `SignInFinished`.

Reply:
Nice idea, thanks!